include/SH3/system/input_config.hpp

```diff
@@ -0,0 +1,31 @@
+/** @file
+ *  Defines configuration variables for the @ref sh3::system::input_system.
+ */
+#ifndef SH3_SYSTEM_INPUT_CONFIG_HPP_INCLUDED
+#define SH3_SYSTEM_INPUT_CONFIG_HPP_INCLUDED
+
+/**
+ *  Whether the @ref sh3::system::input_system should provide timing support for inputs.
+ *  
+ *  With this enabled, the @ref sh3::system::input_system keeps track what fraction of a frame the actions were active for.
+ *  
+ * @note Doesn't work right with SDL timestamps.
+ *       I think the event times are not when the input *happened*, but when the event-loop *received* them, which is not useful to us.
+ */
+#ifdef DOXYGEN
+#define INPUT_PROVIDE_TIMING 0
+#else
+#undef INPUT_PROVIDE_TIMING
+#endif
+/**
+ *  Whether the @ref sh3::system::input_system should provide counting support for inputs.
+ *  
+ *  With this enabled, the @ref sh3::system::input_system keeps track how many times an action was triggered within a frame.
+ */
+#ifdef DOXYGEN
+#define INPUT_PROVIDE_COUNT 1
+#else
+#define INPUT_PROVIDE_COUNT
```

Comment:
For the code it's just relevant which of them is `#define`d. The value doesn't matter at all.
However, you can't document `#undef`ined variables with Doxygen, so for the documentation I just `#define` them to `1` or `0` depending on whether the features are enabled or not. In the actual build, the enabled features are simply `#define`d symbols and the disabled ones are not.